src/services/services.ts

```diff
@@ -1680,7 +1680,7 @@ module ts {
             useCaseSensitiveFileNames: () => false,
             getCanonicalFileName: fileName => fileName,
             getCurrentDirectory: () => "",
-            getNewLine: () => "\r\n"
+            getNewLine: () => (sys && sys.newLine) || "\r\n"

```

Comment:
Calculate this once by introducing `compilerHostNewLine` - also, can you declare `compilerHost` and `program` with `let` instead of `var`?